### PR TITLE
Enable ZSTD compression for binlog

### LINKED ISSUE #15561: [Feature]: Implement binlog compression on minio
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

We've see many people complain about Milvus 2.0 spent too many disk spaces.

One thing we can do is do compress the binlogs and check if we can get some compression ratio, If so compression should also help the load speed due to saved IOs

### Describe the solution you'd like.

_No response_

### Describe an alternate solution.

_No response_

### Anything else? (Additional Context)

_No response_

### PR DESCRIPTION
**Pull request by yah01**
Enable ZSTD compression of Parquet, for milvus' binlog
Resolve #15561 
Signed-off-by: yah01 <yang.cen@zilliz.com>

### CODE REVIEW

**Comment by yah01** [comment]
The compress ratio is:
2.5 for level 1
3.3 for level 3

**Comment by yah01** [comment]
rerun ut

**Comment by xiaofan-luan** [comment]
let's take compression_level(3)

File: .github/workflows/code-checker.yaml [hunk 1]
```diff
@@ -109,5 +109,6 @@ jobs:
           if [[ ! -d "/var/tmp/ccache" ]];then
             mkdir -p /var/tmp/ccache
           fi
+          brew update
```
Comment (jennyli-z) [inline .github/workflows/code-checker.yaml:112]: As we have verified `"-Dzstd_SOURCE=BUNDLED"` is the key to fix the code checker on mac, so please revert this back ,thanks. https://github.com/milvus-io/milvus/runs/5531761338?check_suite_focus=true

File: .golangci.yml [hunk 2]
```diff
@@ -52,5 +52,5 @@ issues:
   max-same-issues: 0
 
 service:
-  golangci-lint-version: 1.27.0 # use the fixed version to not introduce new linters unexpectedly
+  golangci-lint-version: 1.44.2 # use the fixed version to not introduce new linters unexpectedly
```
Comment (jennyli-z) [inline .golangci.yml:55]: As we have verified `"-Dzstd_SOURCE=BUNDLED"` is the key to fix the code checker on mac, so please revert this back ,thanks. https://github.com/milvus-io/milvus/runs/5531761338?check_suite_focus=true

**Pull request merged**